### ISSUE TITLE
Get rid of `REQUIRE_NODE_MODULES`

### ISSUE THREAD

**Issue by smoelius** [trailofbits/necessist#583]
> However, for the long term, I am wondering if we can remove the `REQUIRE_NODE_MODULES` and simply rely on the presence of the `package.json` to decide if we should install node packages or not. There should be only rare cases in which the `package.json` exists and is not required to build or test the project.

_Originally posted by @tarunbhm in https://github.com/trailofbits/necessist/issues/580#issuecomment-1630134892_

**Comment by smoelius** [comment]
@nowak-token Thank you for commenting on this issue.

TBH, I've considered closing this issue. The reason is this. A repository could contain a package.json file just by happenstance. In such a case, it would be weird if Necessist ran `npm install` just because it found the file.

To paint a really bleak picture, imagine a Rust repo with a package.json in the root. Most people use the repo and it seems fine. But someone runs Necessist and it downloads malware via NPM.

Honestly, the more I think about this, the less I'm convinced it's something we should do.

Still, I'd like to give @tarunbhm a chance to comment on the issue he didn't open. :P

**Comment by tarunbhm** [comment]
Yes, looking at the recent incidents, the malicious NPM package installation is a real risk. I think we can close this issue as removing the flag is not going to improve the experience much.

**Comment by smoelius** [comment]
Thanks, @tarunbhm.

@nowak-token I'm going to close the issue. But nonetheless, thank you, sincerely for your interest. 🙏
